routes/common.js

```diff
@@ -4,7 +4,36 @@
 
 var message = require('../models/message');
 var config = require('../config/config');
+var messageCollectionEmitter = require('../EventEmitters/mongoEventsEmitter');
 
+/**
+ * Async save new message to Mongo "Message" collection
+ * @param newMsg - new message to be saved in the mongoDB. the message should be constructed to match message Schema.
+ * @param onFailure - callback to be used in-case save fails
+ * @param onSuccess - callback to be used in-case a successful result is returned
+ */
+function saveNewMessageToMongoDB(newMsg, onFailure, onSuccess){
+    if (newMsg.createdAt) {
+        var err = new Error("Given message shouldn't contain createdAt path!");
+        //return is used to finish function's execution
```

Comment:
A comment should have blank line above it.
If the line above it "new block", there is no need to so.
For example, if there is a comment after "if" statement, we shouldn't add empty line.